release.py

```diff
@@ -177,18 +183,66 @@ def push() -> None:
             raise typer.Exit(result)
 
 
-def check_preconditions() -> None:
+def ensure_all_committed_files() -> Optional[List[str]]:
```

Comment:
I know that the function name is pretty explicit, but a docstring could help if someone is new to the project 😉

Reply:
Maybe you're starting to be a bit too focus on docstrings?

I could write
```
def ensure_all_committed_files() -> Optional[List[str]]:
"""
Ensure all files are committed
"""
```

But that feels really like over-commenting the code, which is also a drawback...